reference/enterprise-api-ref/ldap-api.markdown

```diff
@@ -0,0 +1,125 @@
+---
+layout: default
+title: LDAP authentication API
+published: true
+tags: [reference, enterprise, API, authenticating, LDAP]
+---
+
+LDAP authentication API allows to check ldap user credentials and change LDAP settings.
+
+## Login
+
+**URI:** https://hub.cfengine.com/ldap/login
+
+**Method:** POST
+
+**Parameters:**
+
+* **username** *(string)*
+    Username from LDAP
+* **password** *(string)*
+    User password
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "success": true,
+    "message": "You are successfully authenticated"
+}
+```
+
+## Get settings
+
+**URI:** https://hub.cfengine.com/ldap/settings
+
+**Method:** GET
+
+**Headers:**
+
+* **Authorization: api_token** *(string)*
+    Set token to access api
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "success": true,
+    "data": {
+        "domain_controllers": "local.loc",
+        "base_dn": "DC=local,DC=loc",
+        "login_attribute": "cn",
+        "port": 365,
+        "use_ssl": false,
+        "use_tls": false,
+        "timeout": 5,
+        "admin_username": "cn=admin,DC=local,DC=loc",
+        "admin_password": "Password is set"
+    }
+}
+```
+
+**Output:**
+
+* **domain_controllers**
+    The domain controllers option is   server name located on your network that serve Active Directory.
```

Comment:
Are we allowed to have multiple domain controllers? The name makes me think there can be more than one. Is this stubbed in for future support?

Reply:
New API gives us the possibility to set more than one domain controller, but current UI implementation isn't. So I cant rename domain_controllers to domain_controller.

Reply:
If the field can have multiple then we can leave it. Ui Support isn't important for the API imho

Reply:
Also, I  I've implemented saving the setting only with one domain, so, or we should change the behavior of saving, or rename this attribute.

Reply:
I think it is better to rename to domain_controller
because even if we can save multiple domains via LDAP library API, our API and UI  won't be able to show and save this.